MonoGame.Framework/Audio/SoundEffect.OpenAL.cs

```diff
@@ -125,45 +125,44 @@ private void PlatformLoadAudioStream(Stream s)
 #endif
         }
 
-        private void PlatformInitialize(byte[] buffer, int sampleRate, AudioChannels channels)
+        private void PlatformInitializePCM(byte[] buffer, int offset, int count, int sampleRate, AudioChannels channels, int loopStart, int loopLength)
         {
-			Rate = (float)sampleRate;
-            Size = (int)buffer.Length;
+            Rate = (float)sampleRate;
+            Size = (int)count;
+            Format = channels == AudioChannels.Stereo ? ALFormat.Stereo16 : ALFormat.Mono16;
 
-#if OPENAL && !(MONOMAC || IOS)
-
-            _data = buffer;
-            Format = (channels == AudioChannels.Stereo) ? ALFormat.Stereo16 : ALFormat.Mono16;
-
-#endif
-
-#if MONOMAC || IOS
-
-            //buffer should contain 16-bit PCM wave data
-            short bitsPerSample = 16;
-
-            if ((int)channels <= 1)
-                Format = bitsPerSample == 8 ? ALFormat.Mono8 : ALFormat.Mono16;
-            else
-                Format = bitsPerSample == 8 ? ALFormat.Stereo8 : ALFormat.Stereo16;
-
-            _name = "";
-            _data = buffer;
-
-#endif
+            // Make our own copy of the data.
+            _data = new byte[count];
+            Array.Copy(buffer, offset, _data, 0, count);
```

Comment:
This wasn't here before, but it seems dangerous to not copy the passed in sound buffer as the caller could alter it after the fact.  In fact I expect under XNA it does something similar copying the passed in buffer into the native code.